src/components/BitcoinAmountInput.tsx

```diff
@@ -61,6 +61,78 @@ const BitcoinAmountInput = forwardRef(
           : undefined
     }, [field, inputType])
 
+    const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+      setInputType({
+        type: 'text',
+        inputMode: 'decimal',
+      })
+
+      let displayValue = String(field.value?.value || '')
+      if (isValidNumber(field.value?.value)) {
+        displayValue = formatBtcDisplayValue(field.value!.value!)
+      }
+
+      form.setFieldValue(
+        field.name,
+        {
+          ...field.value,
+          displayValue,
+        },
+        false,
+      )
+      field.onBlur(e)
+    }
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const rawUserInputOrEmpty = e.target.value ?? ''
+      const validNumberRegex = /^-?\d*\.?\d*$/
+      if (!validNumberRegex.test(rawUserInputOrEmpty)) {
+        return
+      }
```

Comment:
I am not sure about this regex test. Shouldn't the next line catch invalid number inputs (`!isValidNumber`)? What would be the impact of removing it?

Reply:
The regex test was introduced because removing the `type` attribute allows users to enter any string. This check quickly identifies invalid numbers before further processing. The `isValidNumber` function is used to check if the entered value is a valid number, but it only accepts `number | undefined | null` as input. Since the value from e.target.value is a `string`, we use parseFloat to convert it.

However, parseFloat has issues, such as interpreting 1.0 as 1, which affects how we handle BTC vs. sats. The regex helps ensure that only valid numeric strings are passed to parseFloat, preventing such issues.

Removing the regex test would mean parseFloat could incorrectly process some inputs, and it would allow users to enter any string, leading to inconsistent behavior.

Reply:
I know it's quite cumbersome, but I think reverting it back will bring `up/down arrows`, but the problem of `1.0` or in `lang="IT"` `1,0` will persist.